### PR TITLE
Add Publisher Action 

### PR DESCRIPTION
**Pull request by hrittikhere**
GitHub Action to automate CRON Tweets from feed! I have sanitized cron on `.github/workflows/publisher-action.yml` to stop the workflow from running for now. The workflow was successful as we can see in this artifact:  https://github.com/hrittikhere/feedposter/runs/6474764539?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/publisher-action.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Publisher
+
+# on: push
+
+# on:
```
Comment (hrittikhere) [inline .github/workflows/publisher-action.yml]: Commented out `on` to not trigger workflow as of now.

File: platforms/twitter.go [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+func AuthenticateTwitter() *twitter.Client {
 
-	const ConsumerKey = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
-	const ConsumerSecret = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
-	const AccessToken = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
-	const AccessSecret = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+	ConsumerKey := os.Getenv("CONSUMER_KEY")
+	ConsumerSecret := os.Getenv("CONSUMER_SECRET")
+	AccessToken := os.Getenv("ACCESS_TOKEN")
+	AccessSecret := os.Getenv("ACCESS_TOKEN_SECRET")
```
Comment (senthilrch) [inline platforms/twitter.go:29]: Add a check to check if the variables filled from ENV are not empty.

**Review by senthilrch — CHANGES_REQUESTED** [review]
Add a check to check if the variables filled from ENV are not empty.

**Comment by hrittikhere** [comment]
We now support #️⃣ for the tags. The request `Dev.to blogs have hashtags. Is there a way to extract those hashtags and add it to the tweet?` is completed🌞

image
image

Known issue: Maybe (PostTitle, PostAuthor, hashtags, PostLink) together can move past the tweet limit. We need to check if that's the case and truncate it.
Next PR will take care of it as well as: 
1. Twitter ID of the author if present
2. YAML Reads for feeds
3. We need to start creating TestCases and I need some help on that end with how to create user facing docs.

File: platforms/twitter.go (hunk 2 above)
Comment (hrittikhere) [reply platforms/twitter.go:29]: ` os.LookupEnv` used and checks added. 
image

**Review by senthilrch — APPROVED** [review]
LGTM! Good job Hrittik...

**Pull request merged**